Tasks/JavaToolInstaller/task.json

```diff
@@ -0,0 +1,155 @@
+            "required": true,
+            "helpMarkDown": "A number that specifies the JDK version to make available on the path."
+        },
+        {
+            "name": "jdkSource",
+            "type": "pickList",
+            "label": "JDK source",
+            "required": true,
+            "options": {
+                "AzureStorage": "Azure Storage",
+                "LocalDirectory": "Local Directory"
+            },
+            "helpMarkDown": "Source for the compressed JDK."
+        },
+        {
+            "name": "jdkPath",
+            "type": "filepath",
+            "label": "JDK path",
+            "required": true,
+            "visibleRule": "jdkSource == LocalDirectory",
+            "helpMarkDown": "Path to where the compressed JDK is located. The path could be in your source repository or a local path on the agent."
+        },
+        {
+            "name": "ConnectedServiceNameARM",
+            "type": "connectedService:AzureRM",
+            "label": "Azure Subscription",
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Choose the Azure Resource Manager subscription for the JDK."
+        },
+        {
+            "name": "storageAccountName",
+            "type": "pickList",
+            "label": "Storage Account Name",
+            "required": true,
+            "properties": {
+                "EditableOptions":"True"
+            },
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Azure Classic and Resource Manager storage accounts are listed. Select the Storage account name in which the JDK is located."
+        },
+        {
+            "name": "containerName",
+            "type": "string",
+            "label": "Container Name",
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Name of the container in the storage account in which the JDK is located."
+        },
+        {
+            "name": "commonVirtualPath",
+            "type": "string",
+            "label": "Common Virtual Path",
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Path to the JDK inside the Azure storage container."
+        },
+        {
+            "name": "fileType",
+            "type": "pickList",
+            "label": "Compressed file type",
+            "options": {
+                "zip": ".zip",
+                "7z": ".7z",
+                "tar": ".tar",
+                "compressedTar": ".tar.gz"
+            },
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Compression type of the JDK."
+        },
+        {
+            "name": "destinationFolder",
+            "type": "filePath",
+            "label": "Destination folder",
+            "defaultValue": "",
+            "required": true,
```

Comment:
Can we make this optional and have a smart default? Prefer "Directory" over "Folder" e.g. "jdkDestinationDirectory"